pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -967,6 +999,7 @@ func (nc *Controller) tryUpdateNodeHealth(node *v1.Node) (time.Duration, v1.Node
 				status:                   &node.Status,
 				probeTimestamp:           nc.nodeHealthMap[node.Name].probeTimestamp,
 				readyTransitionTimestamp: nc.now(),
+				lease:                    observedLease,
```

Comment:
All the changes in this file look reasonable to me.

@gmarek - would you be able to take a look too?